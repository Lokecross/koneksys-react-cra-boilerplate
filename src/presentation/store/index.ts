import reduxStore from './redux/store';

export { reduxStore };
