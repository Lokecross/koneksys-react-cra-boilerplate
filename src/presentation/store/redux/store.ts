import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';

import { useDispatch, useSelector } from 'react-redux';

import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';

import storage from 'redux-persist/lib/storage';

import { rootReducer } from './features/rootReducer';

const persistedReducer = persistReducer(
  {
    key: 'root',
    version: 1,
    storage,
    whitelist: [],
  },
  rootReducer,
);

const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }),
});

const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector = () => useSelector((state: RootState) => state);

export { store, persistor };
