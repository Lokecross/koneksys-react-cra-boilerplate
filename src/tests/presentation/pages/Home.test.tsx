import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { Home } from '@pages';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from '@hooks';
import { store } from '@store/redux/store';
import { QueryClient, QueryClientProvider } from 'react-query';

export const createMockStore = () => ({
  ...store,
  persistor: {
    persist: () => null,
  },
});
const queryClient = new QueryClient();
test('Renders Home page and match to previous snapshot', () => {
  const mockStore = createMockStore();
  // jest.mock('@hookstate/core', () => {
  //   return jest.fn(() => 1);
  // });
  const component = shallow(
    <QueryClientProvider client={queryClient}>
      <ReduxProvider store={mockStore}>
        <ThemeProvider>
          <Home />
        </ThemeProvider>
      </ReduxProvider>
    </QueryClientProvider>,
  );
  expect(component).toMatchSnapshot();
});
