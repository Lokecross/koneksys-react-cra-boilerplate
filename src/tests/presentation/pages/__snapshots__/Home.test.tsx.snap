// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home page and match to previous snapshot 1`] = `
<ContextProvider
  value={false}
>
  <ContextProvider
    value={
      QueryClient {
        "defaultOptions": Object {},
        "mutationCache": MutationCache {
          "config": Object {},
          "listeners": Array [],
          "mutationId": 0,
          "mutations": Array [],
        },
        "mutationDefaults": Array [],
        "queryCache": QueryCache {
          "config": Object {},
          "listeners": Array [],
          "queries": Array [],
          "queriesMap": Object {},
        },
        "queryDefaults": Array [],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "persistor": Object {
            "persist": [Function],
          },
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <ThemeProvider>
        <Home />
      </ThemeProvider>
    </Provider>
  </ContextProvider>
</ContextProvider>
`;
